### ISSUE TITLE
Pd console does not show errors when opening patch from the command line

### ISSUE THREAD

**Issue by Spacechild1** [pure-data/pure-data#2300]
If I open a Pd patch from the command line (e.g. `pd ~/foo/bar.pd`), the Pd console won't show any errors, e.g. when an object could not be found.

The reason seems to be that in Pd 0.55-test1 the GUI is started up *after* patches have been loaded, so the error messages never make it to the GUI.

**Comment by millerpuckette** [comment]
Yikes, that's a show-stopper.  It looks like fixing that will require eitehr (1) caching printout while waiting for the GUI to kick in or (2) actually starting the TCL/TK process early but not sending "vis" to any  patches until the font sizes are known.
But in other situations (starting Pd on a headless system and later connecting to it to debug something) you'd really want the printout to have been cached somewhere - so I've always been wondering what to do with printout while Pd is in invisible state.

It might be reasonable to have separate options as in  "pd -nogui" (meaning we'll never want one) and "pd -invis" (to delay starting GUI until requested) - in the latter case we'd allocate the to-gui buffer and allow printout to go to it whether or not Pd is currently visible.

**Comment by millerpuckette** [comment]
hopefully fixed now.

**Comment by Spacechild1** [comment]
Can't test at the moment, but judging from the recent mails on the Pd list, the problem seems to be still there...

**Comment by millerpuckette** [comment]
tf - I could swear I had that fixed but it's still broken over here too.  I'll get on it.

**Comment by reduzent** [comment]
I tried df7e86a0db4ac76927575c03d9f93f9b4eecf39f and I see now all messages in the console. However, the order of the messages has changed:

Current Pd-console output:
```
WARNING: non-standard samplerate detected: 48000 Hz
netpd uses 44100 Hz by convention. Not following this convention has unwanted side effects and prevents audio samples or live audio from being shared with peers.
CHAT: rdz: <<<<<<<<<<<<<connected<<<<<<<<<<<<<<<<<
iemlib (1.22) library loaded!   (c) Thomas Musil Nov 13 2018 : 20:19:12
   musil@iem.at iem KUG Graz Austria
iemnet - networking with Pd: [tcpclient]
        version 0.3.0
        compiled on Jun 25 2020 at 21:46:11
        copyright © 2010-2015 IOhannes m zmoelnig, IEM
        based on mrpeach/net, based on maxlib
slipdec: maximum packet length is 65536
slipenc: maximum packet length is 65536
```

Output to terminal when using `-stderr` (resembles output of previous Pd versions):
```
iemlib (1.22) library loaded!   (c) Thomas Musil Nov 13 2018 : 20:19:12
   musil@iem.at iem KUG Graz Austria
iemnet - networking with Pd: [tcpclient]
        version 0.3.0
        compiled on Jun 25 2020 at 21:46:11
        copyright © 2010-2015 IOhannes m zmoelnig, IEM
        based on mrpeach/net, based on maxlib
slipdec: maximum packet length is 65536
slipenc: maximum packet length is 65536
WARNING: non-standard samplerate detected: 48000 Hz
netpd uses 44100 Hz by convention. Not following this convention has unwanted side effects and prevents audio samples or live audio from being shared with peers.
CHAT: rdz: <<<<<<<<<<<<<connected<<<<<<<<<<<<<<<<<
```

I don't know if this is considered a problem, I just post my observations.

**Comment by reduzent** [comment]
From what I understand by having followed the mailing list, the issue is related to the GUI being loaded later compared to earlier versions of Pd. Apparently, Patches loaded at startup are now loaded _before_ the GUI is ready. The proposed solution for addressing the  issue discussed here was to cache all printed messages for later printing, which addresses specifically the printing issue. However, loading patches ahead of  GUI probably has more side-effects. See also #2310. Wouldn't it make more sense to delay the loading of startup patches until GUI is ready?

**Comment by umlaeute** [comment]
it seems that this has been fixed for the Pd-0.55-0 release.
can we close this?
